core/primitives/src/runtime/parameter_table.rs

```diff
@@ -15,32 +20,29 @@ pub(crate) enum InvalidConfigError {
     UnknownParameter(#[source] strum::ParseError, String),
     #[error("Unable to parse value: `{1}`")]
     ValueParseError(#[source] serde_json::Error, String),
-    #[error("Parameter name and value must be separated by a `:`")]
-    NoSeparator,
+    #[error("Parameter name and value must be separated by a `:`. Line {0}: `{1}`")]
```

Comment:
A convention is that the error Display implementations should be output a non-capitalized sentence. That allows the description to be inserted into another sentence without it looking super out-of-place. Particularly common examples are `error: {}` and `caused by: {}`, but more complex formatting is also a thing, so it is always worth it to take a moment to consider what these strings would look like when incorporated in a broader sentence.

For this particular the wording could be something like

```
expected a `:` separator between name and value of a parameter `{1}` on line {0}
```

or maybe

```
name and value of the parameter `{1}` on line {0} must be separated by a `:`
```

Sorry to nit here again, my bad for not elaborating on this last time I made the suggestions.

Reply:
Ah now I understand :) Thanks for taking the time to explain. I changed all messages to conform with this idea